smartsim/_core/utils/telemetry/collector.py

```diff
@@ -124,7 +121,7 @@ def __init__(self, entity: JobEntity, sink: Sink) -> None:
         :param sink: destination to write collected information
         """
         super().__init__(entity, sink)
-        self._client: t.Optional[redisa.Redis[bytes]] = None
+        self._client: None
```

Comment:
I'm wondering what would be a better way of making sure we update this in the future in the case of it not being totally reworked/rewritten.  What do we think about throwing an error for these ``self._cleint = None``?

Reply:
added Mock